Connect to api

gasangw opened the pull request:
- Read the [Bookstore API documentation](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) to learn how to use the API.
- Use [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) for making API calls.
- Fetch book list from Bookstore API on Books page load.
- Add an Action Creator, that returns a function.
- Fetch list of books.
- Dispatch another action (add an action type) that will pass API response to reducer and updates the state.
- Refactor your add book and remove book features to persist your changes in the server.
- Change your existing Action Creators, so they return functions.
- Make changes by using API.
- Dispatch an action (action type that existed before refactor) that will updates the state accordingly.
- Add book and remove book should work in the same way after the refactor.
 - No styling is required.

gasangw committed:
Install axio, fetch book using API generated

gasangw committed:
Create actions for adding book, fetching book and removing a book

gasangw committed:
Add delete functionality of the book

rloterh commented on src/redux/configureStore.js:
```diff
@@ -1,4 +1,5 @@
+import logger from 'redux-logger';
 import booksReducer from './books/books';
 import categoriesReducer from './categories/categories';
 
```
- [x] You did a nice job for adding _reduxjs/toolkit_ and _thunk_ packages, however, this project requires that you use _redux_ _toolkit_ to create _thunk action creators_ by using `createAsyncThunk`, similar to the following code example:
```javascript
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios'; 


export const getBooks = createAsyncThunk(
  'books/getBooks',
  async () => {
    const response = await axios.get(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
    );
    return response.data;
  },
);
```

You may also refer to the [redux toolkit](https://redux-toolkit.js.org/introduction/getting-started) page for the same.

rotimiazeez replied:
- As pointed out by the first reviewer using `createAsyncThunk` is part of the project requirement, kindly implement it before submitting again 😉

rloterh commented on src/redux/books/books.js:
```diff
@@ -1,45 +1,55 @@
-import { v4 as uuidv4 } from 'uuid';
+/* eslint-disable camelcase */
```
- [x] You have **disabled linter check** on your code. Please enable **all** linter check on your code to ensure that your code is robust code and clean.

rloterh reviewed (changes requested):
Hi Thomas, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Axios fetch properly implemented


## Required Changes ♻️
_Kindly check the comments under the review._


<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

gasangw committed:
Implement the required changes

rotimiazeez reviewed (changes requested):
Hi @gasangw 👍🏽, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

gasangw commented:
@rotimiazeez i implement it maybe u didnt see it but i implemented in the books/books file.

rloterh reviewed (approved):
## Status: Approved :heavy_check_mark:

Hi Thomas,  

Your project is complete! There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉🌟 

## Highlights 👍
- API fetch properly implemented
- State management properly implemented


<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

gasangw commented:
hello. I implemented what you are asking me to do. you are checking from
the wrong file. The action that fetches the books is in the books/books
where I implemented it from. the file u are commenting from is supposed to
have the  createAsyncThunk

On Fri, 22 Jul 2022 at 17:08, Azeez Rotimi Arigbabuwo <
***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> Hi <STUDENT_MENTION>,
>
> Great work on making the changes 👏
> You've done well implementing some of the requested changes, but there are
> still some which aren't addressed yet.
> Suggested changes
>
> *Check the comments under the review.*
>
> Cheers and Happy coding!👏👏👏
>
> Feel free to leave any questions or comments in the PR thread if something
> is not 100% clear.
> Please, remember to tag me in your question so I can receive the
> notification.
>
> *Please, do not open a new Pull Request for re-reviews. You should use the
> same Pull Request submitted for the first review, either valid or invalid
> unless it is requested otherwise.*
> ------------------------------
>
> In src/redux/configureStore.js
> <https://github.com/gasangw/bookstore/pull/4#discussion_r927745374>:
>
> > @@ -1,4 +1,5 @@
>
>  import { configureStore } from ***@***.***/toolkit';
>
> +import logger from 'redux-logger';
>
>  import booksReducer from './books/books';
>
>  import categoriesReducer from './categories/categories';
>
>
>
>
>
>    - As pointed out by the first reviewer using createAsyncThunk is part
>    of the project requirement, kindly implement it before submitting again
>    😉
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/gasangw/bookstore/pull/4#pullrequestreview-1048041596>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXVLZSBPWMCC5GN5R7HG5S3VVK2NPANCNFSM54KYWB6Q>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
